src/test/java/com/synacy/lesson02/exercises/EnrollmentServiceTest.java

```diff
@@ -0,0 +1,116 @@
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+/**
+ * Created by kenichigouang on 5/3/17.
+ */
+
+@RunWith(MockitoJUnitRunner.class)
+public class EnrollmentServiceTest {
+    private EnrollmentService enrollmentService;
+
+    @Mock StudentProfileService studentProfileService;
+
+    @Mock CourseClassService courseClassService;
+
+    @Mock EnrollmentNotificationService enrollmentNotificationService;
+
+    @Mock StudyLoadFormatter studyLoadFormatter;
+    @Mock SystemService systemService;
+    @Before
+    public void setup() {
+        enrollmentService = new EnrollmentService();
+
+        enrollmentService.setStudentProfileService(studentProfileService);
+        enrollmentService.setCourseClassService(courseClassService);
+        enrollmentService.setEnrollmentNotificationService(enrollmentNotificationService);
+        enrollmentService.setStudyLoadFormatter(studyLoadFormatter);
+        enrollmentService.setSystemService(systemService);
+    }
+
+    @Test
+    public void processEnrollment_shouldPrintStudyLoad() throws Exception {
+        EnrollmentDto enrollmentDetails = mock(EnrollmentDto.class);
+        Printable studyLoadPrintable = mock(StudyLoad.class);
+
+        when(studyLoadFormatter.format(enrollmentDetails)).thenReturn((StudyLoad) studyLoadPrintable);
+
+        enrollmentService.processEnrollment(enrollmentDetails);
+
+        verify(systemService, times(1)).print(
+                ArgumentMatchers.eq(studyLoadPrintable)
+        );
+    }
+
+    @Test
+    public void processEnrollment_shouldNotifyStudentsOnEnrollment() throws Exception {
+
+        EnrollmentDto enrollmentDetails = mock(EnrollmentDto.class);
+
+        enrollmentService.processEnrollment(enrollmentDetails);
+
+        verify(enrollmentNotificationService, times(1)).emailStudent(
+                ArgumentMatchers.eq(StudentEmailType.ENROLLMENT),
+                ArgumentMatchers.eq(enrollmentDetails)
+        );
+
+    }
+
+    @Test
+    public void processEnrollment_shouldEnrollStudentInClasses() throws Exception {
+
+        EnrollmentDto enrollmentDetails = mock(EnrollmentDto.class);
+        Student student = mock(Student.class);
+
+        when(enrollmentDetails.getStudent()).thenReturn(student);
+
+        Set<CourseClass> enrolledClasses = new HashSet<>();
+        enrolledClasses.add(mock(CourseClass.class));
+        enrolledClasses.add(mock(CourseClass.class));
+        enrolledClasses.add(mock(CourseClass.class));
+
+        when(enrollmentDetails.getEnrolledClasses()).thenReturn(enrolledClasses);
+
+
+        Iterator<CourseClass> enrolledClass = enrolledClasses.iterator();
+
+        enrollmentService.processEnrollment(enrollmentDetails);
+
+        verify(courseClassService, times(1)).enrollStudentToClass(
+                ArgumentMatchers.eq(student),
+                ArgumentMatchers.eq(enrolledClass.next())
+        );
```

Comment:
We are only verifying one method call with the first `CourseClass` object as parameter. How about the rest `CourseClass` objects in the set?